lib/core-integration/src/Test/Integration/Scenario/API/Wallets.hs

```diff
@@ -1471,13 +1498,532 @@ spec = do
+                    (Link.getWallet @'Byron sourceWallet)
+                    Default
+                    Empty >>= flip verify
+                    [ expectField (#balance . #available) (.> Quantity 0)
+                    ]
+
+            targetWallet <- emptyWallet ctx
+            let payload = Json [json|{"passphrase": #{fixturePassphrase}}|]
+            let ep = Link.migrateWallet sourceWallet targetWallet
+            r <- request @[ApiTransaction n] ctx ep Default payload
+            let srcId = sourceWallet ^. walletId
+            verify r
+                [ expectResponseCode @IO HTTP.status403
+                , expectErrorMessage (errMsg403NothingToMigrate srcId)
+                ]
+
+    it "BYRON_MIGRATE_03 - \
+        \actual fee for migration is the same as the predicted fee."
+        $ \ctx -> forM_ [fixtureRandomWallet, fixtureIcarusWallet] $ \fixtureByronWallet -> do
+            -- Restore a Byron wallet with funds.
+            sourceWallet <- fixtureByronWallet ctx
+
+            -- Request a migration fee prediction.
+            let ep0 = (Link.getMigrationInfo sourceWallet)
+            r0 <- request @ApiByronWalletMigrationInfo ctx ep0 Default Empty
+            verify r0
+                [ expectResponseCode @IO HTTP.status200
+                , expectField #migrationCost (.> Quantity 0)
+                ]
+
+            -- Perform the migration.
+            targetWallet <- emptyWallet ctx
+            let payload = Json [json|{"passphrase": #{fixturePassphrase}}|]
+            let ep1 = Link.migrateWallet sourceWallet targetWallet
+            r1 <- request @[ApiTransaction n] ctx ep1 Default payload
+            verify r1
+                [ expectResponseCode @IO HTTP.status202
+                , expectField id (`shouldSatisfy` (not . null))
+                ]
+
+            -- Verify that the fee prediction was correct.
+            let actualFee = fromIntegral $ sum $ apiTransactionFee
+                    <$> getFromResponse id r1
+            let predictedFee =
+                    getFromResponse (#migrationCost . #getQuantity) r0
+            actualFee `shouldBe` predictedFee
+
+    it "BYRON_MIGRATE_04 - migration fails with a wrong passphrase"
+        $ \ctx -> forM_ [fixtureRandomWallet, fixtureIcarusWallet] $ \fixtureByronWallet -> do
+        -- Restore a Byron wallet with funds, to act as a source wallet:
+        sourceWallet <- fixtureByronWallet ctx
+
+        -- Perform a migration from the source wallet to a target wallet:
+        targetWallet <- emptyWallet ctx
+        r0 <- request @[ApiTransaction n] ctx
+            (Link.migrateWallet sourceWallet targetWallet )
+            Default
+            (Json [json|{"passphrase": "not-the-right-passphrase"}|])
+        verify r0
+            [ expectResponseCode @IO HTTP.status403
+            , expectErrorMessage errMsg403WrongPass
+            ]
+
+    it "NETWORK_SHELLEY - Wallet has the same tip as network/information" $
```

Comment:
This one had to be moved to Wallets.hs? It rather belongs to Network.hs...

Reply:
Yes / No. I moved it out of `Network` because it actually does make check on a given wallet that has to be a Shelley wallet (which we don't have on the cardano-node integration). Why in `Wallets` then? Because it's still fetching for the wallet's tip and comparing that to the network's tip, so in that sense, both are a good fit (and to be honest, I didn't feel like creating an additional `Scenario/Api/ByronNetwork` because, it's kinda confusing.. the network endpoints are no different between byron and shelley..)

Reply:
ok, makes sense. :+1: